.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,237 @@
+            gc: boehm
+            os: ubuntu-latest
+            target-platform: library
+          - scala: 2.12.12
+            build-mode: debug
+            gc: immix
+            os: ubuntu-latest
+            target-platform: library
+          - scala: 2.11.12
+            build-mode: debug
+            gc: commix
+            os: ubuntu-latest
+            target-platform: library
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Calculate binary version
+        run: |
+          version=${{ matrix.scala }}
+          echo "binary-version=${version%.*}" >> $GITHUB_ENV
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/coursier
+            ~/.ivy2/cache
+            ~/.sbt/boot
+          key: ${{ runner.os }}-deps-${{ env.binary-version }}-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/build.properties') }}
+
+      - name: Cache docker
+        uses: actions/cache@v1
+        with:
+          path: /tmp/docker-registry
+          key: docker-registry-${{ hashFiles('ci-docker/Dockerfile') }}
+
+      - name: Run tests
+        env:
+          SCALANATIVE_MODE: ${{ matrix.build-mode }}
+          SCALANATIVE_GC: ${{ matrix.gc }}
+          SCALA_VERSION: ${{ matrix.scala}}
+          SCALANATIVE_OPTIMIZE: true
+          TEST_COMMAND: test-runtime
+          TARGET_DOCKER_PLATFORM: ${{ matrix.target-platform }}
+        run: ./ci-docker/run-test-gha.sh ${{ needs.build-image.outputs.image-name }}
+
+  test-runtime-no-opt:
+    name: Test runtime no-opt
+    runs-on: ${{ matrix.os }}
+    needs: [tests-tools, build-image]
+    strategy:
+      fail-fast: false
+      matrix:
+        scala: [ 2.13.3, 2.12.12, 2.11.12 ]
+        build-mode: [ debug, release-fast ]
+        gc: [ immix ]
+        os: [ ubuntu-latest ]
+        target-platform: [ library ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Calculate binary version
+        run: |
+          version=${{ matrix.scala }}
+          echo "binary-version=${version%.*}" >> $GITHUB_ENV
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/coursier
+            ~/.ivy2/cache
+            ~/.sbt/boot
+          key: ${{ runner.os }}-deps-${{ env.binary-version }}-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/build.properties') }}
+
+      - name: Cache docker
+        uses: actions/cache@v1
+        with:
+          path: /tmp/docker-registry
+          key: docker-registry-${{ hashFiles('ci-docker/Dockerfile') }}
+
+      - name: Run tests
+        env:
+          SCALANATIVE_MODE: ${{ matrix.build-mode }}
+          SCALANATIVE_GC: ${{ matrix.gc }}
+          SCALA_VERSION: ${{ matrix.scala}}
+          SCALANATIVE_OPTIMIZE: false
```

Comment:
This seems to be the only difference between `test-runtime` and `test-runtime-no-opt`. Why not use an additional parameter `opt` next to `gc` and `build-mode` instead?

Reply:
In Travis CI there were 6 no-opt tests in total debug + release-fast with immix GC for each version. 
Using this here would produce 18 tests (3 versions * 3 GC * 2 debug modes). 
We can add them using `include` clause but it would take much place in yaml since we need to set each parameter in the matrix. Same if we would use `exclude` as when creating holes in `test-runtime`. Also, each change in the matrix parameters list, like a new field, would result in the need to fix all of them. 

Although this duplication might look redundant in my opinion it's easier to maintain in the longer term.